Library/Traceparent.ts

```diff
@@ -0,0 +1,121 @@
+import Util = require("./Util");
+import CorrelationIdManager = require("./CorrelationIdManager");
+
+/**
+ * Helper class to manage parsing and validation of traceparent header. Also handles hierarchical
+ * back-compatibility headers generated from traceparent. W3C traceparent spec is documented at
+ * https://www.w3.org/TR/trace-context/#traceparent-field
+ */
+class Traceparent {
+    private static DEFAULT_TRACE_FLAG = "00";
+    private static DEFAULT_VERSION = "00";
+
+    public legacyRootId: string;
+    public parentId: string;
+    public spanId: string;
+    public traceFlag: string = Traceparent.DEFAULT_TRACE_FLAG;
+    public traceId: string;
+    public version: string = Traceparent.DEFAULT_VERSION;
+
+    constructor(traceparent?: string, parentId?: string) {
+        if (traceparent) { // traceparent constructor
+            // If incoming request contains traceparent: parse it, set operation, parent and telemetry id accordingly. traceparent should be injected into outgoing requests. request-id should be injected in back-compat format |traceId.spanId. so that older SDKs could understand it.
+            if (traceparent.split(",").length > 1) { // If more than 1 traceparent is present, discard both
+                this.traceId = Util.w3cTraceId();
```

Comment:
I know this is added before this PR, the name is weird, especially when you look at `Util.w3cTraceId().substr(0, 16)`.